packages/@aws-cdk/aws-cloudwatch-actions/lib/ec2.ts

```diff
@@ -0,0 +1,47 @@
+import * as cloudwatch from '@aws-cdk/aws-cloudwatch';
+import { Stack } from '@aws-cdk/core';
+
+// keep this import separate from other imports to reduce chance for merge conflicts with v2-main
+// eslint-disable-next-line no-duplicate-imports, import/order
+import { Construct } from '@aws-cdk/core';
+
+/**
+ * Types of EC2 actions available
+ */
+export enum Ec2InstanceActions {
+  /**
+   * Stop the instance
+   */
+  STOP = 'stop',
+  /**
+   * Terminatethe instance
+   */
+  TERMINATE = 'terminate',
+  /**
+   * Recover the instance
+   */
+  RECOVER = 'recover',
+  /**
+   * Reboot the instance
+   */
+  REBOOT = 'reboot'
+}
+
+/**
+ * Use an EC2 action as an Alarm action
+ */
+export class Ec2Action implements cloudwatch.IAlarmAction {
+  private ec2Action: string;
+
+  constructor(instanceAction: Ec2InstanceActions) {
+    this.ec2Action = instanceAction;
+  }
+
+  /**
+   * Returns an alarm action configuration to use an EC2 action as an alarm action
+   */
+  bind(_scope: Construct, _alarm: cloudwatch.IAlarm): cloudwatch.AlarmActionConfig {
+    return { alarmActionArn: `arn:aws:automate:${Stack.of(_scope).region}:ec2:${this.ec2Action}` };
```

Comment:
Are we sure this doesn't need any permissions added to any role to make this work?

I can tell the documentation on this feature is very sparse. I honestly can't see how this is supposed to work, or where the targeted instance is supposed to be coming from.

That latter detail, what the instance is, seems like a good detail to include in the docstring of this class as well, perhaps together with an example of how/why you would use this action.

Reply:
I was trying to create a stack with this feature and found the ec2 Actions to be missing. I will test that actually this works, discuss the details here and document the conclusions in the README file

Reply:
That sounds fantastic, thanks!

Reply:
Tested, works OK: I created an instance and disabled its network interface what caused status check to fail. This triggered an alarm that forced the instance to reboot and voilà! instance recovered.
[Here is](https://gist.github.com/cyuste/2908b285bd8d9b0fa0689ec2d90183d6) a snippet of the code that I used for the test. I also added a SNS action to compare between them. No fancy roles or anything required.
But (and it's a big but) this works because the alarm is configured using an EC2 per-instance metric. So actually the test I had in the last commit should fail and the correct test would be something like this:
```ts
test('cannot use a non-EC2 alarm with EC2 action', () => {
  // GIVEN
  const stack = new Stack();
  const alarm = new cloudwatch.Alarm(stack, 'Alarm', {
    metric: new cloudwatch.Metric({ namespace: 'AWS', metricName: 'Henk' }),
    evaluationPeriods: 3,
    threshold: 100,
  });
  expect(() => {
    alarm.addAlarmAction(new actions.Ec2Action(actions.Ec2InstanceAction.REBOOT));
  }).toThrow('You must use an alarm related to an EC2 Per-Instance Metric');
});
```

I'm trying to find a way to detect this inside cw-actions, but I haven't found it yet. I'll also see if I can add this check in the cloudwatch Alarm class

Reply:
🙏

Reply:
I finally implemented it in cloudwatch package and added a test to validate that I cannot bind an EC2 action (based on the arn) to an alarm that is not instance related.
Based on a few tests in the aws console I saw that I can only bind an EC2 action when the alarm is based on a single, per-instance metric (for example, I cannot use a combination of an instance and a sns metric or even two metrics from the same instance)
I implemented this check in a new `addAlarmAction` method in  `alarm` class and left the original `addAlarmAction` method from `alarm-base` as it is used by `composite-alarm` class